### DISCUSSION TITLE
[lint] `ts` files imported as `js` files

### DISCUSSION THREAD

**Discussion by Conaclos** [rome/tools#1639] [Suggestions]
TypeScript Compiler does not automatically add js file's extension upon transpiration. When the module's output is ESM, this leads to invalid imports. The code cannot be natively executed on nodejs. To tackle this issue, the TypeScript Compiler enables to import ts files using js extension.

For instance, the following code is valid:

```ts
@filename a.ts
export function f(){}

@filename b.ts
import {f} from "./a.js" // using js extension to import a ts file
```

This enables to natively execute the transpiled code. I'm using this technique in my TypeScript projects.

Using Rome v10.0.7-beta.8, `rome check` outputs `Cannot find ./a.js from b.ts`.

Related issue: #1632.

**Comment by ematipico** [comment]
This is something we want to address, for sure. At the moment our resolution logic is poor, that's why we decided to turn it off for the upcoming release.
